File: szkopul/poi26_lic3/lic.py
from liclib import IleGier, KimJestem, DajCiag, DajM, DajK, RuchPartnera, Licytuj

def main():
    for _ in range(IleGier()):
        moj_ciag = DajCiag()
        m = DajM()
        k = DajK()

        if KimJestem() == 0:
            # Gramy jako Algosia.
            stawka = 0
            for i in range(3):
                stawka += 1  # Algosia licytuje wartość o 1 większą.
                if i == 2:
                    Licytuj(stawka, 1)  # Algosia kończy grę.
                    break
                Licytuj(stawka, 0)  # Algosia licytuje wartość stawki i nie kończy gry.
                stawka, koniec = RuchPartnera()  # Wczytujemy ruch Bajtka.
        else:
            # Gramy jako Bajtek.
            while True:
                stawka, koniec = RuchPartnera()  # Wczytujemy ruch Algosii.
                if koniec:
                    break  # Bajtek kończy grę.
                Licytuj(stawka + 1, 0)  # Bajtek licytuje wartość o 1 większą i nie kończy gry.

if __name__ == "__main__":
    main()

File: szkopul/poi26_lic3/lic.cpp
#include "liclib.h"

int main() {
    for (int i = 0; i < IleGier(); ++i) {
        std::vector<int> moj_ciag = DajCiag();
        [[maybe_unused]] int m = DajM();
        [[maybe_unused]] int k = DajK();
        int stawka = 0;
        int num_iters = 3;

        if (KimJestem() == 0) {
            // Gra Algosi.
            for (int i = 0; i < num_iters; i++) {
                if (i == num_iters - 1) {
                    Licytuj(stawka + 1, true); // Algosia kończy grę.
                    break;
                }
                Licytuj(stawka + 1, false); // Algosia licytuje.
                auto [nowa_stawka, koniec] = RuchPartnera();
                stawka = nowa_stawka;
            }
        } else {
            // Gra Bajtka.
            while (true) {
                auto [nowa_stawka, koniec] = RuchPartnera();
                stawka = nowa_stawka;
                if (koniec) {
                    break; 
                }
                Licytuj(stawka + 1, false); // Bajtek licytuje wartość o 1 większą i nie kończy gry.
            }
        }
    }

    return 0;
}
